crates/transport_iroh/src/tests/mod.rs

```diff
@@ -49,387 +46,3 @@ fn validate_plain_relay_url() {
     assert!(result.is_err());
     assert!(format!("{result:?}").contains("disallowed plaintext relay url"));
 }
-
-struct MockTxHandler {
-    /// Mock function to implement the new_listening_address() method of the
-    /// TxBaseHandler trait
-    pub current_url: Arc<Mutex<Url>>,
-    /// Mock function to implement the recv_space_notify() method of the
-    /// TxSpaceHandler trait
-    pub recv_space_notify: Arc<
-        dyn Fn(Url, SpaceId, bytes::Bytes) -> K2Result<()>
-            + 'static
-            + Send
-            + Sync,
-    >,
-}
-
-impl std::fmt::Debug for MockTxHandler {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("MockTxHandler").finish()
-    }
-}
-
-impl TxBaseHandler for MockTxHandler {
-    fn new_listening_address(&self, this_url: Url) -> BoxFut<'static, ()> {
-        *self.current_url.lock().unwrap() = this_url;
-        Box::pin(async {})
-    }
-
-    fn peer_connect(&self, _peer: Url) -> K2Result<()> {
-        Ok(())
-    }
-
-    fn peer_disconnect(&self, _peer: Url, _reason: Option<String>) {}
-}
-
-impl TxHandler for MockTxHandler {
-    fn preflight_gather_outgoing(
-        &self,
-        _peer_url: Url,
-    ) -> BoxFut<'_, K2Result<bytes::Bytes>> {
-        Box::pin(async { Ok(Bytes::new()) })
-    }
-
-    fn preflight_validate_incoming(
-        &self,
-        _peer_url: Url,
-        _data: bytes::Bytes,
-    ) -> BoxFut<'_, K2Result<()>> {
-        Box::pin(async { Ok(()) })
-    }
-}
-
-impl TxSpaceHandler for MockTxHandler {
-    fn recv_space_notify(
-        &self,
-        peer: Url,
-        space_id: SpaceId,
-        data: bytes::Bytes,
-    ) -> K2Result<()> {
-        (self.recv_space_notify)(peer, space_id, data)
-    }
-
-    fn set_unresponsive(
-        &self,
-        _peer: Url,
-        _when: Timestamp,
-    ) -> BoxFut<'_, K2Result<()>> {
-        Box::pin(async { Ok(()) })
-    }
-
-    fn are_all_agents_at_url_blocked(&self, _peer_url: &Url) -> K2Result<bool> {
-        Ok(false)
-    }
-}
-
-#[tokio::test]
-async fn connect_two_endpoints() {
```

Comment:
The two integration tests here are just adapted to use the harness and moved to the `integration.rs` file.